### PR TITLE
Add more error diffusion methods

### LINKED ISSUE #27: Add more error diffusion methods
These still need references.

**Shiau-Fan:**
```
      *  4
1  1  2         (1//8)
```

**Shiau-Fan 2:**
```
         *  8 
1  1  2  4      (1//16)
```

**Zhigang Fan:**
```   
   *  7
1  3  5         (1//16)

### PR DESCRIPTION
**Pull request by adrhill**
Closes #27 by adding the error diffusion methods `Fan`, `ShiauFan`, `ShiauFan2` and  `FalseFloydSteinberg` (the latter as an unexported easter egg).

### CODE REVIEW

File: src/error_diffusion.jl [hunk 1]
```diff
@@ -183,3 +182,67 @@ Error diffusion algorithm using the filter
...
+
+Error diffusion algorithm using the filter
+```
+*   3
+3   2         (1//16)
+```
+Occasionally, you will see this filter erroneously called the Floyd-Steinberg filter.
+
+# Note
+There is no reason to use this algorithm, which is why DitherPunk doesn't export it.
```
Comment (johnnychen94) [inline src/error_diffusion.jl:244]: lol

File: src/error_diffusion.jl [hunk 2]
```diff
@@ -183,3 +182,67 @@ Error diffusion algorithm using the filter
...
+      *  7
+1  3  5            (1//16)
+```
+A modification of the weights used in the Floyd-Steinberg algorithm.
+
+[1] Z. Fan, "A Simple Modification of Error Diffusion Weights",
+    IS&T's 46th Annual Conference, May 9-14, 1993, Final Program and Advanced Printing of
+    Paper Summaries, pp 113-115 (1993).
+"""
+Fan() = ErrorDiffusion(OffsetMatrix([0 0 0 7; 1 3 5 0]//16, 0:1, -2:1))
```
Comment (johnnychen94) [inline src/error_diffusion.jl:200]: The name `Fan` can be very ambiguous and is very likely to cause name conflict when used together with other ecosystems. Maybe `Fan93`?

**Review by johnnychen94 — APPROVED** [review]
You're more than an engineer! Really like your output.

File: src/error_diffusion.jl (hunk 2 above)
Comment (adrhill) [reply src/error_diffusion.jl:200]: Good point!

**Pull request merged**